### PR TITLE
Remove useless table directories when use HiveCatalog.dropTable and purge is true

### PR DESCRIPTION
**Pull request by wypb**
related to [#1764](https://github.com/apache/iceberg/issues/1764)

### CODE REVIEW

File: core/src/main/java/org/apache/iceberg/CatalogUtil.java [hunk 1]
```diff
@@ -90,6 +94,31 @@ public static void dropTableData(FileIO io, TableMetadata metadata) {
...
+        HadoopFileIO hadoopFileIO = (HadoopFileIO) io;
+        Configuration conf = hadoopFileIO.conf();
+        Path path = new Path(tableBaseLocation);
+        FileSystem fs = Util.getFs(path, conf);
+
+        // Because it might be a partitioned table, there might be a partition path below the base path,
+        // so we need to delete recursively
+        fs.delete(path, true);
+      } else {
+        io.deleteFile(tableBaseLocation);
```
Comment (rdblue) [inline core/src/main/java/org/apache/iceberg/CatalogUtil.java:115]: I don't think this is correct. The base location isn't a file, so `deleteFile` should not be used. Instead, I think this should log an error that there was no way to delete the base location and it was skipped.

File: hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java [hunk 2]
```diff
@@ -168,6 +168,7 @@ public boolean dropTable(TableIdentifier identifier, boolean purge) {
 
       if (purge && lastMetadata != null) {
         CatalogUtil.dropTableData(ops.io(), lastMetadata);
+        CatalogUtil.dropTableBaseLocation(ops.io(), lastMetadata.location());
```
Comment (rdblue) [inline hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java:171]: I don't think this should be the default behavior. It is fine if people want to do this in catalogs, but I would prefer to have an option to turn it on or off.

File: core/src/main/java/org/apache/iceberg/CatalogUtil.java (hunk 1 above)
Comment (397090770) [reply core/src/main/java/org/apache/iceberg/CatalogUtil.java]: Hi @rdblue Thank you for you reply. I have updated the code based on the comments above, can you help me look again.

**Comment by rdblue** [comment]
@397090770, I've added this to the agenda for our next community sync. I think we should talk about when Iceberg tables own the storage location and when they should automatically delete a location or prefix.

**Comment by 397090770** [comment]
@rdblue Yes, you are right, we should define these behaviors clearly. By the way, can you give me a link to the agenda, thank you.

**Comment by rdblue** [comment]
The agenda for the next sync is in the [Iceberg sync doc](https://docs.google.com/document/d/1YuGhUdukLP5gGiqCbk0A5_Wifqe2CZWgOd3TbhY3UQg/edit#heading=h.gx5jk191pfuv).

**Comment by gaborkaszab** [comment]
Hey @rdblue ,
I came into this topic recently as well where I expected Iceberg's dropTable() to also drop the whole directory structure for a table but apparently it doesn't. I checked a number of related issue tickets and unfinished PRs, also I checked the Iceberg sync doc. As I see the community came to a conclusion that both ways are valid (owning and not owning the whole directory structure) and there should be a flag either on table or catalog level to indicate which approach to use.

However, I haven't seen any code changes after this conclusion was made. Is there a specific reason (e.g. the community still not sure about this functionality) or is it just no one took the effort to actually implement it? I wonder if I could revive the conversation about this topic.

**Pull request closed without merging**